read_test.go

```diff
@@ -392,6 +392,135 @@ func TestReadNativeFrames(t *testing.T) {
 	}
 }
 
+func TestReadNativeFrames_BytePixels(t *testing.T) {
+	cases := []struct {
```

Comment:
Can we add a test case to test for the condition where `vl%2 != 0` and assert that we get an expectedError we want? We would probably also want to have some kind of sentinel error we return to make checking this easy (e.g. kinda like `ErrorUnsupportedBitsAllocated`